src/controller/gc/controller.go

```diff
@@ -0,0 +1,208 @@
+	Count(ctx context.Context, query *q.Query) (int64, error)
+	// GetSchedule get the current gc schedule
+	GetSchedule(ctx context.Context) (*scheduler.Schedule, error)
+	// CreateSchedule create the gc schedule with cron string
+	CreateSchedule(ctx context.Context, cron string, parameters map[string]interface{}) (int64, error)
+	// DeleteSchedule remove the gc schedule
+	DeleteSchedule(ctx context.Context) error
+}
+
+// NewController creates an instance of the default repository controller
+func NewController() Controller {
+	return &controller{
+		taskMgr:      task.NewManager(),
+		exeMgr:       task.NewExecutionManager(),
+		schedulerMgr: scheduler.New(),
+	}
+}
+
+type controller struct {
+	taskMgr      task.Manager
+	exeMgr       task.ExecutionManager
+	schedulerMgr scheduler.Scheduler
+}
+
+// Start starts the manual GC
+func (c *controller) Start(ctx context.Context, parameters map[string]interface{}) (int64, error) {
+	execID, err := c.exeMgr.Create(ctx, gcVendorType, -1, task.ExecutionTriggerManual, parameters)
+	if err != nil {
+		return -1, err
+	}
+	taskID, err := c.taskMgr.Create(ctx, execID, &task.Job{
+		Name: job.ImageGC,
+		Metadata: &job.Metadata{
+			JobKind: job.KindGeneric,
+		},
+		Parameters: parameters,
+	})
+	if err != nil {
+		return -1, err
+	}
+	defer func() {
+		if err == nil {
+			return
+		}
+		if err := c.taskMgr.Stop(ctx, taskID); err != nil {
+			log.Errorf("failed to stop the task %d: %v", taskID, err)
+		}
+	}()
+	return execID, nil
+}
+
+// Stop ...
+func (c *controller) Stop(ctx context.Context, taskID int64) error {
+	return c.taskMgr.Stop(ctx, taskID)
+}
+
+// GetLog ...
+func (c *controller) GetLog(ctx context.Context, executionID int64) ([]byte, error) {
+	tasks, err := c.taskMgr.List(ctx, q.New(q.KeyWords{"ExecutionID": executionID}))
+	if err != nil {
+		return nil, err
+	}
+	if len(tasks) == 0 {
+		return nil, errors.New(nil).WithCode(errors.NotFoundCode).WithMessage("no gc task is found")
+	}
+	return c.taskMgr.GetLog(ctx, tasks[0].ID)
+}
+
+// Count ...
+func (c *controller) Count(ctx context.Context, query *q.Query) (int64, error) {
+	query.Keywords["VendorType"] = gcVendorType
+	return c.exeMgr.Count(ctx, query)
+}
+
+// GetGCByID ...
+func (c *controller) GetByID(ctx context.Context, id int64) (*History, error) {
+	exe, err := c.exeMgr.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	tasks, err := c.taskMgr.List(ctx, q.New(q.KeyWords{"ExecutionID": exe.ID}))
+	if err != nil {
+		return nil, err
+	}
+	if len(tasks) == 0 {
+		return nil, errors.New(nil).WithCode(errors.NotFoundCode).WithMessage("no gc task is found")
+	}
+
+	extraAttrsString, err := json.Marshal(exe.ExtraAttrs)
+	if err != nil {
+		return nil, err
+	}
+	return &History{
+		ID:         exe.ID,
+		Name:       gcVendorType,
+		Kind:       exe.Trigger,
+		Parameters: string(extraAttrsString),
+		Deleted:    false,
+		Schedule: Schedule{Schedule: &ScheduleParam{
+			Type: exe.Trigger,
+		}},
+		Status:       tasks[0].Status,
```

Comment:
Can we just use the properties of execution so that we don't need to search the task here?
Seems the execution doesn't contain `UpdateTime`, we can open an issue and add it